src/sagemaker/config/config_utils.py

```diff
@@ -0,0 +1,170 @@
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This file contains util functions for the sagemaker Defaults Config.
+
+These utils may be used inside or outside the config module.
+"""
+from __future__ import absolute_import
+
+import logging
+import sys
+
+
+def get_sagemaker_config_logger():
+    """Return a logger with the name 'sagemaker.config'
+
+    If the logger to be returned has no level or handlers set, this will get level and handler
+    attributes. (So if the SDK user has setup loggers in a certain way, that setup will not be
+    changed by this function.) It is safe to make repeat calls to this function.
+    """
+    sagemaker_config_logger = logging.getLogger("sagemaker.config")
+    sagemaker_logger = logging.getLogger("sagemaker")
+
+    if sagemaker_config_logger.level == logging.NOTSET:
+        sagemaker_config_logger.setLevel(logging.INFO)
+
+    # check sagemaker_logger here as well, so that if handlers were set for the parent logger
+    # already, we dont change behavior for the child logger
+    if len(sagemaker_config_logger.handlers) == 0 and len(sagemaker_logger.handlers) == 0:
+        # use sys.stdout so logs dont show up with a red background in a notebook
+        handler = logging.StreamHandler(sys.stdout)
+
+        formatter = logging.Formatter("%(name)s %(levelname)-4s - %(message)s")
+        handler.setFormatter(formatter)
+        sagemaker_config_logger.addHandler(handler)
+
+        # if a handler is being added, we dont want the root handler to also process the same events
+        sagemaker_config_logger.propagate = False
+
+    return sagemaker_config_logger
+
+
+def _log_sagemaker_config_single_substitution(source_value, config_value, config_key_path: str):
+    """Informs the SDK user whether a config value was present and automatically substituted
+
+    Args:
+        source_value: The value that will be used if it exists. Usually, this is user-provided
+            input to a Class or to a session.py method, or None if no input was provided.
+        config_value: The value fetched from sagemaker_config. If it exists, this is the value that
+            will be used if direct_input is None.
+        config_key_path: A string denoting the path of keys that point to the config value in the
+            sagemaker_config.
+
+    Returns:
+        None. Logs information to the "sagemaker.config" logger.
+    """
+    logger = get_sagemaker_config_logger()
+
+    if config_value is not None:
+
+        if source_value is None:
+            # Sagemaker Config value is going to be used
+            logger.info(
+                "Applied value\n  config key = %s\n  config value that will be used = %s",
+                config_key_path,
+                config_value,
+            )
```

Comment:
Would something like this be better?
```
            # It may not be safe to log ARNs to stdout by default. If DEBUG logs were enabled
            # by the user though, we can include more diagnostic info
            if logger.isEnabledFor(logging.DEBUG):
                logger.debug(
                    "Applied value\n  config key = %s\n  config value that will be used = %s",
                    config_key_path,
                    config_value,
                )
            else:
                logger.info(
                    "Applied value from config key = %s",
                    config_key_path,
                )
```